env0/provider.go

```diff
@@ -127,7 +128,13 @@ func configureProvider(version string, p *schema.Provider) schema.ConfigureConte
 		if err != nil {
 			return nil, diag.Diagnostics{diag.Diagnostic{Severity: diag.Error, Summary: err.Error()}}
 		}
+		apiClient := client.NewApiClient(httpClient)
+		if apiClient == nil {
+			return nil, diag.Diagnostics{diag.Diagnostic{Severity: diag.Error, Summary: "Failed to fetch organiztions."}}
+		}
 
-		return client.NewApiClient(httpClient), nil
+		apiClient.Organization()
```

Comment:
Since you're already calling this method.
You could set the `cachedOrganizationId` in `ApiClient` struct.

With that said, I do wonder how it will affect some acceptance tests. (I guess we'll wait and see).

Reply:
good thought :) I'll run test as soon as I know it fixes the issue, currently when testing I can still see multiple numbers of invocation of authorizer lambda

Reply:
@TomerHeber this one solve the issue I had, and also addresses your suggestion :)
[5049f75](https://github.com/env0/terraform-provider-env0/pull/456/commits/5049f757d2745b15506f444ad1998c7764e163ff)